porcupine/_logs.py

```diff
@@ -87,11 +78,18 @@ def setup(verbose_logger: str | None) -> None:
     if sys.stderr is not None:
         # not running in pythonw.exe, can also show something in terminal
         print_handler = logging.StreamHandler(sys.stderr)
-        if verbose_logger is None:
-            print_handler.setLevel(logging.WARNING)
-        else:
-            print_handler.setLevel(logging.DEBUG)
-            print_handler.addFilter(_FilterThatDoesntHideWarnings(verbose_logger))
+        print_handler.setLevel(logging.DEBUG)
+        if not all_loggers_verbose:
+            print_handler.addFilter(
+                lambda record: (
+                    # Always show warnings and errors
+                    record.levelno >= logging.WARNING
+                    # If --verbose-loggers=foo is passed on command line, show all messages
+                    # from loggers "foo", "foo.bar", "foo.bar.baz"
+                    or record.name in verbose_loggers
+                    or any(record.name.startswith(name + ".") for name in verbose_loggers)
```

Comment:
This looks like it could be slow. This runs for all log messages, and this would probably be best avoided with a `setLevel` when there are no extra-verbose loggers.